hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -64,1271 +64,1639 @@
+
+        auto existing_connection =
+          active_connections_.get(response.member_uuid);
+        if (existing_connection) {
+            connection->close(
+              (boost::format(
+                 "Duplicate connection to same member with uuid : %1%") %
+               boost::uuids::to_string(response.member_uuid))
+                .str());
+            return existing_connection;
+        }
 
-            void ClientConnectionManagerImpl::trigger_cluster_reconnection() {
-                if (reconnect_mode_ == config::client_connection_strategy_config::reconnect_mode::OFF) {
-                    HZ_LOG(logger_, info, "RECONNECT MODE is off. Shutting down the client.");
-                    shutdown_with_external_thread(client_.get_hazelcast_client_implementation());
-                    return;
-                }
+        auto new_cluster_id = response.cluster_id;
+        boost::uuids::uuid current_cluster_id = cluster_id_;
+
+        HZ_LOG(logger_,
+               finest,
+               boost::str(boost::format(
+                            "Checking the cluster: %1%, current cluster: %2%") %
+                          new_cluster_id % current_cluster_id));
+
+        auto cluster_id_changed = !current_cluster_id.is_nil() &&
+                                  !(new_cluster_id == current_cluster_id);
+        if (cluster_id_changed) {
+            HZ_LOG(
+              logger_,
+              warning,
+              boost::str(
+                boost::format(
+                  "Switching from current cluster: %1%  to new cluster: %2%") %
+                current_cluster_id % new_cluster_id));
+            client_.get_hazelcast_client_implementation()->on_cluster_restart();
+        }
 
-                if (client_.get_lifecycle_service().is_running()) {
-                    submit_connect_to_cluster_task();
-                }
-            }
+        auto connections_empty = active_connections_.empty();
+        active_connection_ids_.put(connection->get_connection_id(), connection);
+        active_connections_.put(response.member_uuid, connection);
+        if (connections_empty) {
+            cluster_id_ = new_cluster_id;
+            fire_life_cycle_event(
+              lifecycle_event::lifecycle_state::CLIENT_CONNECTED);
+        }
 
-            std::shared_ptr<Connection> ClientConnectionManagerImpl::get_random_connection() {
-                if (smart_routing_enabled_) {
-                    auto member = load_balancer_.next_(client_.get_cluster());
-                    if (!member) {
-                        return nullptr;
-                    }
-                    auto connection = get_connection(member->get_uuid());
-                    if (connection) {
-                        return connection;
-                    }
-                }
+        auto local_address = connection->get_local_socket_address();
+        if (local_address) {
+            HZ_LOG(
+              logger_,
```

Comment:
Is there a problem with MACRO formatting?

Reply:
I don't know what do you mean by problem? Is it inconsistent? The way it works is that if a function call is too long to fit into the column limit, it puts every parameter into a different line.